test/test_datasets.py

```diff
@@ -1763,5 +1662,96 @@ def inject_fake_data(self, tmpdir, config):
+        ("train-challenge", False): "data_large_challenge",
+        ("val", False): "val_large",
+        ("train-standard", True): "data_256_standard",
+        ("train-challenge", True): "data_256_challenge",
+        ("val", True): "val_256",
+    }
+    # (class, idx)
+    _CATEGORIES_CONTENT = (
+        ("/a/airfield", 0),
+        ("/a/apartment_building/outdoor", 8),
+        ("/b/badlands", 30),
+    )
+    # (file, idx)
+    _FILE_LIST_CONTENT = (
+        ("Places365_val_00000001.png", 0),
+        *((f"{category}/Places365_train_00000001.png", idx)
+          for category, idx in _CATEGORIES_CONTENT),
+    )
+
+    @staticmethod
+    def _make_txt(root, name, seq):
+        file = os.path.join(root, name)
+        with open(file, "w") as fh:
+            for text, idx in seq:
+                fh.write(f"{text} {idx}\n")
+
+    @staticmethod
+    def _make_categories_txt(root, name):
+        Places365TestCase._make_txt(root, name, Places365TestCase._CATEGORIES_CONTENT)
+
+    @staticmethod
+    def _make_file_list_txt(root, name):
+        Places365TestCase._make_txt(root, name, Places365TestCase._FILE_LIST_CONTENT)
+
+    @staticmethod
+    def _make_image(file_name, size):
+        os.makedirs(os.path.dirname(file_name), exist_ok=True)
+        PIL.Image.fromarray(np.zeros((*size, 3), dtype=np.uint8)).save(file_name)
+
+    @staticmethod
+    def _make_devkit_archive(root, split):
+        Places365TestCase._make_categories_txt(root, Places365TestCase._CATEGORIES)
+        Places365TestCase._make_file_list_txt(root, Places365TestCase._FILE_LISTS[split])
+
+    @staticmethod
+    def _make_images_archive(root, split, small):
+        folder_name = Places365TestCase._IMAGES[(split, small)]
+        image_size = (256, 256) if small else (512, random.randint(512, 1024))
+        files, idcs = zip(*Places365TestCase._FILE_LIST_CONTENT)
+        images = [f.lstrip("/").replace("/", os.sep) for f in files]
+        for image in images:
+            Places365TestCase._make_image(os.path.join(root, folder_name, image), image_size)
+
+        return [(os.path.join(root, folder_name, image), idx) for image, idx in zip(images, idcs)]
+
+    def inject_fake_data(self, tmpdir, config):
+        self._make_devkit_archive(tmpdir, config['split'])
+        return len(self._make_images_archive(tmpdir, config['split'], config['small']))
+
+    def test_classes(self):
+        classes = list(map(lambda x: x[0], self._CATEGORIES_CONTENT))
+        with self.create_dataset() as (dataset, _):
+            self.assertEqual(dataset.classes, classes)
+
+    def test_class_to_idx(self):
+        class_to_idx = dict(self._CATEGORIES_CONTENT)
+        with self.create_dataset() as (dataset, _):
+            self.assertEqual(dataset.class_to_idx, class_to_idx)
+
+    def test_images_download_preexisting(self):
+        with self.assertRaises(RuntimeError):
+            with self.create_dataset({'download': True}):
```

Comment:
Nit:

```suggestion
            with self.create_dataset(download=True):
```

Reply:
Using download=True won't work here. That would set the kwargs which populates special_kwargs with key `download=True`. However, it then gets set to False [here](https://github.com/pytorch/vision/blob/97b6b150a79c718be1d3cf163c9b6576a0822bad/test/datasets_utils.py#L371).
Passing it as a dictionary sets config argument, which would not run into this issue.

Reply:
Interesting. I need to follow up on that. Thanks for letting me know.